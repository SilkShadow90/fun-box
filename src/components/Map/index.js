import { connect } from 'react-redux';
import Map from './Map';
import { selectIsSidebarOpen } from '../../data/app';
import {
  selectNavigationList,
  selectPlaceMarkRoute,
  setMapCenter,
  setNewPlaceMarkPosition,
} from '../../data/navigation';

const mapStateToProps = state => ({
  isSidebarOpen: selectIsSidebarOpen(state),
  placeMarkList: selectNavigationList(state),
  placeMarkRoute: selectPlaceMarkRoute(state),
});

const mapDispatchToProps = {
  setMapCenter,
  setNewPlaceMarkPosition,
};

export default connect(mapStateToProps, mapDispatchToProps)(Map);
