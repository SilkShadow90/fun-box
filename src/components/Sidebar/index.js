import { connect } from 'react-redux';
import Sidebar from './Sidebar';
import {
  addNavigationList,
  removeNavigationList,
  reorderNavigationList,
  selectNavigationList,
} from '../../data/navigation';
import { selectIsSidebarOpen, switchSidebar } from '../../data/app';

const mapStateToProps = state => ({
  navigationList: selectNavigationList(state),
  isSidebarOpen: selectIsSidebarOpen(state),
});

const mapDispatchToProps = {
  addNavigationList,
  removeNavigationList,
  reorderNavigationList,
  switchSidebar,
};

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
