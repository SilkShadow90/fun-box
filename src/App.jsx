import React from 'react';
import classNames from 'classnames';

import { Map, Sidebar } from './components';
import './App.css';

const App = () => (
  <div className={classNames('App')}>
    <Sidebar />
    <Map />
  </div>
);

export default App;
