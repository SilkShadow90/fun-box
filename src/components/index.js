import Map from './Map';
import Sidebar from './Sidebar';
import DropDownList from './DropDownList';
import Input from './Input';
import IconButton from './IconButton';

export {
  Map,
  Sidebar,
  DropDownList,
  Input,
  IconButton,
};
