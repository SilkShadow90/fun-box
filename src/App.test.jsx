import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';
import { Sidebar, Map } from './components';

describe('check App', () => {
  const renderer = new ShallowRenderer();

  renderer.render(<App />);
  const result = renderer.getRenderOutput();

  test('check expect type', () => {
    expect(result.type).toBe('div');
  });

  test('check expect className', () => {
    expect(result.props.className).toBe('App');
  });

  test('check expect children', () => {
    expect(result.props.children).toEqual([
      <Sidebar />,
      <Map />,
    ]);
  });

  test('check expect match snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
