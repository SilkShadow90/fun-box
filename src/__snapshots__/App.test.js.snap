// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check App check expect match snapshot 1`] = `
<div
  className="App"
>
  <Connect(Sidebar) />
  <Connect(Map) />
</div>
`;
